newbank/server/NewBankClientHandler.java

```diff
@@ -158,6 +171,30 @@ private void handleRequestCallBack() throws IOException{
 		out.println("--------------------------------------\n");
 	}
 
+	private void handleResetPassword() throws IOException{
+		// ask for user name
+		out.println("Enter username");
+		String userName = in.readLine();
+		while(!bank.isUsername(userName)){
+			out.println("Not a valid username. Please type your username again.");
+			userName = in.readLine();			
+		}
+		// ask for memorable word
+		out.println("What was the name of your first pet?");
+		String memorableWord = in.readLine();
+		out.println("Checking Details...");
+		// check memorable word against word stored
+		if(bank.isMemorableWord(userName, memorableWord)){
+			out.println("Memorable word correct. Please type out a new password.");
+			String password = in.readLine();
+			bank.getCustomer(userName).setPassword(password);
+			out.println("Password reset.");			
+		}
+		// requests
```

Comment:
What's this bit for?

Reply:
we need the customer object to reset password which we get from the userName enterred